elastic/kibana
# [Upgrade Assistant] Delete deprecation log cache

Fixes https://github.com/elastic/kibana/issues/113635

### Summary

This PR changes the behaviour of the CTA from the deprecations count checkpoint so that it calls an API that ES has implemented to suppress logs that are duplicates of existing logs.

**How to test**
1. Start up elasticsearch, `yarn es snapshot --license=trial`
3. Start up Kibana, `yarn start` and navigate to UA
4. Generate a few deprecation issues, at the moment navigating to Observability app will generate a bunch
5. Verify that clicking `Reset counter` calls the delete cache API
6. Verify that if the delete cache API fails, the counter and the callout state remains the same. In order to do this replace [this line](https://github.com/elastic/kibana/pull/114113/files#diff-5c5ee69c2a806117f23ff1a65e442b0544de6b233fd89cc2a072ab3180c8af98R144) with `return response.notFound();`


**Screenshots**

<details>
<summary>Overview of changes</summary>

// Error toast
![Screenshot 2021-10-06 at 16 57 54](https://user-images.githubusercontent.com/1191206/136229351-b9c9212c-37d9-4f94-a50c-6b6102d82d74.png)

</details>
